### PR TITLE
Bug Fixes

### LINKED ISSUE #690: pin creation space field ui bugs
Select 'parent space' wrong name in pin creation
Screenshot_20230702-104557

And once selected, if the space name is long, the field label is squashed: 
Screenshot_20230702-104612

And it allows selecting spaces we aren't allowed to post pins in...

### LINKED ISSUE #747: Creation sheets
news, pins, etc: make CTA buttom for post the highlighted one, not the cancel.

### PR DESCRIPTION
**Pull request by kokodavid**
Pin creation space field UI bugs and Creation sheets CTA button fixes.
Fixes #747
Fixes #690

### CODE REVIEW

File: app/lib/features/events/dialogs/create_event_sheet.dart [hunk 1]
```diff
@@ -257,13 +257,12 @@ class _CreateEventSheetConsumerState extends ConsumerState<CreateEventSheet> {
               : context.goNamed(Routes.main.name),
           child: const Text('Cancel'),
           style: ElevatedButton.styleFrom(
-            backgroundColor: Theme.of(context).colorScheme.neutral,
+            // backgroundColor: _titleInput.isNotEmpty
+            //     ? Theme.of(context).colorScheme.success
+            //     : Theme.of(context).colorScheme.success.withOpacity(0.6),
```
Comment (gnunicorn) [inline app/lib/features/events/dialogs/create_event_sheet.dart:262]: why are we committing commented lines? Please remove.
```suggestion
```

File: app/lib/features/news/pages/simple_post.dart [hunk 2]
```diff
@@ -176,13 +178,12 @@ class _SimpleNewsPostState extends ConsumerState<SimpleNewsPost> {
               : context.goNamed(Routes.main.name),
           child: const Text('Cancel'),
           style: ElevatedButton.styleFrom(
-            backgroundColor: Theme.of(context).colorScheme.neutral,
+            // backgroundColor: _titleInput.isNotEmpty
+            //     ? Theme.of(context).colorScheme.success
+            //     : Theme.of(context).colorScheme.success.withOpacity(0.6),
```
Comment (gnunicorn) [inline app/lib/features/news/pages/simple_post.dart:183]: same.
```suggestion
```

File: app/lib/features/pins/dialogs/create_pin_sheet.dart [hunk 3]
```diff
@@ -200,13 +256,12 @@ class _CreatePinSheetConsumerState extends ConsumerState<CreatePinSheet> {
               : context.goNamed(Routes.main.name),
           child: const Text('Cancel'),
           style: ElevatedButton.styleFrom(
-            backgroundColor: Theme.of(context).colorScheme.neutral,
+            // backgroundColor: _titleInput.isNotEmpty
+            //     ? Theme.of(context).colorScheme.success
+            //     : Theme.of(context).colorScheme.success.withOpacity(0.6),
```
Comment (gnunicorn) [inline app/lib/features/pins/dialogs/create_pin_sheet.dart:261]:
```suggestion
```

File: app/lib/features/pins/dialogs/create_pin_sheet.dart [hunk 4]
```diff
@@ -147,48 +147,104 @@ class _CreatePinSheetConsumerState extends ConsumerState<CreatePinSheet> {
...
+              //         );
+              //         ref.read(selectedSpaceIdProvider.notifier).state =
+              //             newSelectedSpaceId;
+              //       },
+              //     ),
+              //     validator: (x) => (ref.read(selectedSpaceIdProvider) != null)
+              //         ? null
+              //         : 'You must select a space',
+              //   ),
+              // ),
```
Comment (gnunicorn) [inline app/lib/features/pins/dialogs/create_pin_sheet.dart:247]:
```suggestion
```

File: app/lib/features/pins/dialogs/create_pin_sheet.dart [hunk 5]
```diff
@@ -147,48 +147,104 @@ class _CreatePinSheetConsumerState extends ConsumerState<CreatePinSheet> {
...
               ),
               FormField(
-                builder: (state) => ListTile(
-                  title: Text(
-                    _selectedSpace ? 'Space' : 'Please select a space',
-                    style: Theme.of(context).textTheme.bodyMedium,
-                  ),
-                  subtitle: state.errorText != null
+                builder:(state) =>  GestureDetector(
+                  onTap: ()async{
```
Comment (gnunicorn) [inline app/lib/features/pins/dialogs/create_pin_sheet.dart:151]: style

```suggestion
                  onTap: () async {
```

**Review by gnunicorn — CHANGES_REQUESTED** [review]
Thanks, mate.

I have some minor clean up requests, and could you please add a file in `.changes` documenting what you fixed for end-users, like [described here in our docs](https://docs.acter.global/docs/contributing/reviews/#tracking-changes), please!?!

**Comment by kokodavid** [comment]
@gnunicorn 

Thanks for the review , I have made the changes requested and also created the .changes folder and created a file '779-fixes.md' . I just want to confirm if that's the correct way before I push the changes . Kindly let me know if it's correct .
image

**Review by gnunicorn — APPROVED** [review]

**Pull request merged**